dbt/include/bigquery/macros/materializations/incremental.sql

```diff
@@ -27,6 +27,84 @@
   {% do return(strategy) %}
 {% endmacro %}
 
+{% macro get_columns_with_types_in_query(select_sql) %}
+    {% call statement('get_columns_with_types_in_query', fetch_result=True, auto_begin=False) -%}
+        select * from (
+            {{ select_sql }}
+        ) as __dbt_sbq
+        where false
+        limit 0
+    {% endcall %}
+    {%- set result = load_result('get_columns_with_types_in_query') -%}
+    {{ log('result ' ~ result) }}
+    {%- set table = result.table -%}
+    {{ log('table ' ~ table) }}
+    {%- set columns = result.columns -%}
+    {{ log('columns ' ~ columns) }}
+    {{ return(load_result('get_columns_with_types_in_query').table.columns | list) }}
```

Comment:
Hello @jtcohen6 @McKnight-42, I could use some help over here:
To create a table such as 
```
create or replace table `project`.`cou`.`test_ingestion_dbt` (`x` INT64)
```

Yet, as it goes through agate lib apparently, I'm getting
```
create or replace table `project`.`cou`.`test_ingestion_dbt` (`x` <dbt.clients.agate_helper.Number object at 0x104688ee0>)
```

I guess I need to use `get_columns_in_relation` from the adapter so that I get the real BQ type and not a `Number` which can't be used for creating the table.
My intent is to use the `data_type` here https://github.com/dbt-labs/dbt-bigquery/pull/136/files#diff-167e3557df7f18f1520c5db0045dfac9923e38a617c909d703be844192b28ebeR76

Except for that, I'm making progress but we're not there yet (once I have everything working, I'll still need to write tests).
I'm wasting a lot of time working logging fields for dev as I don't think I can work with a step by step debugger (in Python code and in Jinja), isn't it?

Thanks!